libtransact/src/contract/archive/mod.rs

```diff
@@ -256,6 +261,21 @@ fn find_scar<P: AsRef<Path>>(name: &str, version: &str, paths: &[P]) -> Result<P
         })
 }
 
+fn validate_scar_file_name(file_name: &str, version: &str) -> Result<(), Error> {
+    if file_name.contains('_') {
+        return Err(Error::new(&format!(
+            "invalid .scar file name, must not include '_': {}",
+            file_name
+        )));
+    } else if let Err(err) = VersionReq::parse(version) {
+        return Err(Error::new_with_source(
+            &format!("invalid version requirement: `{}`", version),
+            Box::new(err),
+        ));
+    }
```

Comment:
This is already done in the `find_scar` method, so it's not necessary here.

Reply:
Updated